content/blog/astradb-haystack-integration/index.md

```diff
@@ -0,0 +1,175 @@
+
+> Note: if you were running this code in production, you'd want to save these as environment variables to keep things nice and secure.
+
+```python
+from getpass import getpass
+
+OPENAI_API_KEY = getpass("Enter your openAI key:")
+ASTRA_DB_ID = getpass("Enter your Astra database ID:")
+ASTRA_DB_APPLICATION_TOKEN = getpass("Enter your Astra application token (e.g.AstraCS:xxx ):")
+ASTRA_DB_REGION = getpass("Enter your AstraDB Region: ")
+ASTRA_DB_COLLECTION_NAME = getpass("enter your Astra collection name:")
+ASTRA_DB_KEYSPACE_NAME = getpass("Enter your Astra keyspace name:")
+```
+
+## Using the Astra DocumentStore in an index pipeline 
+Next we'll make a Haystack pipeline, create some embeddings from documents, and add them into the [`AstraDocumentStore`](https://docs.haystack.deepset.ai/v2.0/docs/astradocumentstore).
+
+```python
+import logging
+
+from haystack import Document, Pipeline
+
+from haystack.components.embedders import SentenceTransformersDocumentEmbedder, SentenceTransformersTextEmbedder
+from haystack.components.writers import DocumentWriter
+from haystack.document_stores.types import DuplicatePolicy
+
+from astra_haystack.document_store import AstraDocumentStore
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+embedding_model_name = "sentence-transformers/all-MiniLM-L6-v2"
+
+# embedding_dim is the number of dimensions the embedding model supports.
+document_store = AstraDocumentStore(
+    astra_id=ASTRA_DB_ID,
+    astra_region=ASTRA_DB_REGION,
+    astra_collection=ASTRA_DB_COLLECTION_NAME,
+    astra_keyspace=ASTRA_DB_KEYSPACE_NAME,
+    astra_application_token=ASTRA_DB_APPLICATION_TOKEN,
+    duplicates_policy=DuplicatePolicy.SKIP,
+    embedding_dim=384,
+)
+
+
+# Add Documents
+documents = [
+    Document(content="There are over 7,000 languages spoken around the world today."),
+    Document(
+        content="Elephants have been observed to behave in a way that indicates"
+        " a high level of self-awareness, such as recognizing themselves in mirrors."
+    ),
+    Document(
+        content="In certain parts of the world, like the Maldives, Puerto Rico, "
+        "and San Diego, you can witness the phenomenon of bioluminescent waves."
+    ),
+]
+index_pipeline = Pipeline()
+index_pipeline.add_component(
+    instance=SentenceTransformersDocumentEmbedder(model=embedding_model_name),
+    name="embedder",
+)
+index_pipeline.add_component(instance=DocumentWriter(document_store=document_store, policy=DuplicatePolicy.SKIP), name="writer")
+index_pipeline.connect("embedder.documents", "writer.documents")
+
+index_pipeline.run({"embedder": {"documents": documents}})
+
+print(document_store.count_documents())
+```
+If all has gone well, there should be 3 documents. 🎉
+
+## Using the `AstraRetriever` in a Haystack RAG pipeline
```

Comment:
```suggestion
## Use the `AstraRetriever` in a Haystack RAG pipeline
```

Reply:
why "Use" instead of "Using"? Not opposed just curious about your reasoning.

Reply:
In general, gerunds aren't a best practice for technical writing.